File: ex1.py
import numpy as np
import timeit

def mat_prod(A, B):
    A_nrow, A_ncol = A.shape
    B_nrow, B_ncol = B.shape
    
    if A_ncol != B_nrow:
        print("Não é possível multiplicar essas matrizes.")
        return
    
    C = np.zeros((A_nrow, B_ncol))
    for i in range(A_nrow):
        for j in range(B_ncol):
            C[i,j] = sum(A[i,k]*B[k,j] for k in range(A_ncol))
    
    return C

tempo = timeit.timeit("mat_prod(np.random.randint(1, 100,(5,5)), np.random.randint(1, 100,(5,5))),", globals= globals(), number = 100)
print(tempo)

File: ex1.jl
function mat_prod(A, B)
    A_nrow, A_ncol = size(A)
    B_nrow, B_ncol = size(B)

    if A_ncol != B_nrow
        println("A multiplicação não pode ser feita entre as matrizes.")

    else
        C = zeros(A_ncol, B_nrow)
        for i in 1::A_nrow
            for j in 1::B_ncol
            end
        end
    end        
end

A = [1 2; 1 2]
print(A[1,2])